src/pages/index.js

```diff
@@ -0,0 +1,346 @@
+  })
+  .catch(console.error);
+}
+
+previewModalcloseBtn.addEventListener("click", () => {
+  closeModal(previewModal);
+});
+
+let selectedCard, selectedCardId;
+
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardLikedBtn = cardElement.querySelector(".card__like-button");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-button");
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  
+  if (data.isLiked) { 
+    cardLikedBtn.classList.add("card__like-button_liked");
+  }
+
+  cardLikedBtn.addEventListener("click", (evt) => 
+    handleLike(evt, data._id, cardLikedBtn));
+
+  cardImageEl.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalCaptionEl.textContent = data.name;
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+  });
+
+  cardDeleteBtn.addEventListener("click", () => 
+    handleDeleteCard(cardElement, data._id, cardDeleteBtn));
+
+  return cardElement;
+}
+
+ 
+
+function handleDeleteCard(cardElement, cardId, cardDeleteBtn) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+  
+  setButtonText(cardDeleteBtn, true, "Delete", "Deleting...");
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+
+  if (avatarDeletetBtn) {
+    setButtonText(avatarDeletetBtn, true, "Delete", "Deleting...");
+  }
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      
+      closeModal(deleteModal);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      setButtonText(avatarDeletetBtn, false, "Delete", "Deleting...");
+    });
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  //change text content to "Saving...""
+  const cardSubmitBtn = evt.submitter; 
+  // cardSubmitBtn.textContent = "Saving...";
+  setButtonText(cardSubmitBtn, true);
+  console.log(setButtonText);
+
+  const updatedUserInfo = {
+    name: editModalNameInput.value,
+    about: editModalDescriptionInput.value,
+  };
+
+  api
+    .editUserInfo(updatedUserInfo) 
+    .then((data) => {
+      profileName.textContent = data.name; 
+      profileDescription.textContent = data.about; 
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      //Call setButtonText instead
+      setButtonText(cardSubmitBtn, false);
+    });
+}
+
+//TODO - implement loading text for all other form submissions
+
+function handleAddCardSumbmit(evt) {
+  evt.preventDefault();
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  const cardEl = getCardElement(inputValues);
+  cardForm.reset();
+  closeModal(cardModal);
```

Comment:
```js
  cardForm.reset();
  closeModal(cardModal);
```

These 2 lines should be in `.then` in line 254


You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**

You need to **reset** the form only  in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened, and the user can try to send the data again to the server without having to type the data into the inputs once more.